docs: Use of $any() has no runtime incidence.

JeanMeche opened the pull request:
Some users are woried about `$any()` as function calls are to be avoided in templates. This sentence clarifies the situation.

fixes #48781

dylhunn commented on aio/content/guide/template-expression-operators.md:
```diff
@@ -59,6 +59,8 @@ The `$any()` cast function also works with `this` to allow access to undeclared
 
 The `$any()` cast function works anywhere in a binding expression where a method call is valid.
 
+Also note that `$any()` has only an incidence on the typing. There is no method call in the generated code; the `$any()` function is entirely compiled away.
```
"has only an incidence" -> "only effects"

dylhunn committed:
docs: Use of $any() has no runtime incidence.
Some users are woried about `$any()` as function calls are to be avoided in templates. This sentence clarifies the situation.

fixes #48781

jessicajaniuk commented:
This PR was merged into the repository by commit c03e6a6097506da53b9c74b724b6029814a42445.
